### ISSUE TITLE
Unique Cache Keys

### ISSUE THREAD

**Issue by rpf3** [redis/redis-om-node#59]
This is a bit of a contrived example but given the following schema how would you suggest to maintain unique entities?

``` typescript
class BlogPost extends Entity {}

const schema = new Schema(BlogPost, {
  id: { type: 'number' },
  subject: { type: 'string' },
  body: { type: 'string' }
});
```

If my server is handling two concurrent requests and both have a cache miss initially, then ideally I only want to save the blog posts to the cache for one of the requests.  But given that the `IdStrategy` is currently using [ulid](https://github.com/redis/redis-om-node/blob/e8392a527a10b190151fb8e77392bdbd6690013a/lib/schema/schema.ts#L119), I don't see how you'd do something like this.

Normally I'd have posted this question to the GitHub discussions but I noticed that wasn't enabled yet.

**Comment by guyroyse** [comment]
One of the items in my backlog—at an admittedly lower priority—is to add a "keys provided" option to Redis OM so that it doesn't generate the ULID and expects you to set the entity ID. With something like that, you could set the key to something unique in what you are caching and the second entry would just overwrite the first.

In all honestly, I think you could set the `.entityId` property after calling `.create()` now and this would work. But this wasn't something that I designed Redis OM to do. So, no guarantees. TypeScript wouldn't be happy with it by JavaScript would just honey badger it.

**Comment by rpf3** [comment]
So you are envisioning something provided at the model creation itself rather than a new `IdStrategy`.  Just brainstorming but I think it could be cool if the `IdStrategy` function could use the schema definition to generate a unique key for any given entity.  That way you could even do something with compound keys for uniqueness based off of a hash of a few schema fields.

**Comment by guyroyse** [comment]
That's a really good idea. The problem right now is that the ID is generated before there is any data. `.create()` just creates an empty Entity and sets the ID. Probably a way to change that. Off the top of my head, I'm thinking the IdStrategy needs to have an optional argument to be the "seed" used to generate the ID.

**Comment by CaptainCodeman** [comment]
May apply: https://github.com/redis/redis-om-node/issues/40#issuecomment-1100228715

**Comment by Gliba** [comment]
@CaptainCodeman is there a way on new syntax (createEntity)?

**Comment by guyroyse** [comment]
This was fixed with the release of Redis OM 0.4.0 Beta. Install it with `npm install redis-om@beta`.
